packages/app/src/app/overmind/effects/vscode/composeMenuAppTree.ts

```diff
@@ -0,0 +1,77 @@
+import { MenuId } from './Workbench';
+
+export type MenuAppItems = Array<{
+  group: string;
+  order: number;
+  command?: {
+    id: string;
+    title: string;
+    when?: { key: string };
+    precondition?: { key: string };
+    toggled?: { key: string };
+  };
+  title?: string;
+  when?: { key: string };
+  submenu?: MenuAppItems;
+  submenuId?: number;
+}>;
+
+const DEFAULT_ITEMS = [
```

Comment:
Really nice on typing and how you were able to extract the `MenuId` from the `Workbench`!!